### ISSUE TITLE
Ambiguous implicits for ExecutionContext

### ISSUE THREAD

**Issue by github-benoit-dav** [agourlay/cornichon#598]
Hi,

Trying to upgrade from 0.19.9 to 0.20.0, I face the following compile error in some of our features extending CornichonFeature and calling a method requiring an implicit ExecutionContext:

```
... ambiguous implicit values:
[error]  both lazy value ec in trait BaseFeature of type scala.concurrent.ExecutionContext
[error]  and method executionContext in trait AsyncTestSuite of type scala.concurrent.ExecutionContext
[error]  match expected type scala.concurrent.ExecutionContext
```
Example:

```
package my.feature.test

import com.github.agourlay.cornichon.CornichonFeature

import scala.concurrent.duration._
import scala.concurrent.{Await, Future}

class TestFeature extends CornichonFeature {

//  implicit val _ec: ExecutionContext = executionContext
  beforeFeature {
    Await.result(Future.apply(()), 10.seconds)
  }
//...
}
```

This is probably due to [this commit](https://github.com/agourlay/cornichon/commit/7707d4242dd7c7321cb73ff83b0ee70f79713043)

I could define a local implicit to work around this, but as both implicits are defined in the parent traits of CornichonFeature, I though I might be considered as a bug, also I'm not a specialist. Please confirm.

Thanks a lot !
Benoit

**Comment by agourlay** [comment]
Hi Benoit, thanks for the report 👍 

I believe the initial issue is this change during the migration away from Monix https://github.com/agourlay/cornichon/commit/f608fb9999d4bf200f5766ca1b70c8d96de860ae#diff-f556fe79a9fbdce49ee286f2955b17105e91ad326664b5bbc9f6121f8baa2331L29

The Monix's Scheduler was not explicitly exposed as an `ExecutionContext` so it never conflicted with the one from `AsyncTestSuite`  in the `cornichon-scalatest` module.

I have then reintroduced a public `ExecutionContext` in  https://github.com/agourlay/cornichon/commit/7707d4242dd7c7321cb73ff83b0ee70f79713043 while trying to reduce the blast radius of the breaking change. But however this time, the type was explicit which causes a conflict.

In terms of solution I can see

1. remove https://github.com/agourlay/cornichon/commit/7707d4242dd7c7321cb73ff83b0ee70f79713043 and forces users needing an `ExecutionContext` to handle that.
2. remove the `ExecutionContext` from `BaseFeature` and add it only for feature outside of the `cornichon-scalatest` module.

So far I prefer the first solution as it might be surprising for users to see this default `ExecutionContext` provided.

**Comment by agourlay** [comment]
fixed in 0.20.1
